test/test_distributed_hydrostatic_model.jl

```diff
@@ -0,0 +1,113 @@
+    return min(Δx_min, Δy_min)
+end
+
+function solid_body_rotation_test(grid)
+
+    free_surface = SplitExplicitFreeSurface(; substeps = 5, gravitational_acceleration = 1)
+    coriolis     = HydrostaticSphericalCoriolis(rotation_rate = 1)
+
+    model = HydrostaticFreeSurfaceModel(; grid,
+                                        momentum_advection = VectorInvariant(),
+                                        free_surface = free_surface,
+                                        coriolis = coriolis,
+                                        tracers = :c,
+                                        tracer_advection = WENO(),
+                                        buoyancy = nothing,
+                                        closure = nothing)
+
+    g = model.free_surface.gravitational_acceleration
+    R = grid.radius
+    Ω = model.coriolis.rotation_rate
+
+    uᵢ(λ, φ, z) = 0.1 * cosd(φ) * sind(λ)
+    ηᵢ(λ, φ, z) = (R * Ω * 0.1 + 0.1^2 / 2) * sind(φ)^2 / g * sind(λ)
+    cᵢ(λ, φ, z) = Gaussian(λ, φ - 5, 10)
+
+    set!(model, u=uᵢ, η=ηᵢ, c=cᵢ)
+
+    @show Δt_local = 0.1 * Δ_min(grid) / sqrt(g * grid.Lz) 
+    @show Δt = all_reduce(Δt_local, architecture(grid), op = min)
+
+    for _ in 1:10
+        time_step!(model, Δt)
+    end
+
+    return merge(model.velocities, model.tracers, (; η = model.free_surface.η))
+end
+
+
+Nx = 32
+Ny = 32
+
+for arch in archs
+
+    @testset "Testing distributed solid body rotation" begin
+        grid = LatitudeLongitudeGrid(arch, size = (Nx, Ny, 1),
+                                        halo = (3, 3, 3),
+                                        latitude = (-80, 80),
+                                        longitude = (-160, 160),
+                                        z = (-1, 0),
+                                        radius = 1,
+                                        topology=(Bounded, Bounded, Bounded))
+
+        global_grid = reconstruct_global_grid(grid)
+
+        us, vs, ws, cs, ηs = solid_body_rotation_test(global_grid)
+
+        us = Array(interior(us))
+        vs = Array(interior(vs))
+        ws = Array(interior(ws))
+        cs = Array(interior(cs))
+        ηs = Array(interior(ηs))
+
+        @info "  Testing distributed solid body rotation with architecture $arch"
+        u, v, w, c, η = solid_body_rotation_test(grid)
+
+        u = Array(interior(u))
+        v = Array(interior(v))
+        w = Array(interior(w))
+        c = Array(interior(c))
+        η = Array(interior(η))
+
+        @test all(isapprox(u, partition_global_array(arch, us, size(u)), atol=1e-20, rtol = 1e-15))
+        @test all(isapprox(v, partition_global_array(arch, vs, size(v)), atol=1e-20, rtol = 1e-15))
+        @test all(isapprox(w, partition_global_array(arch, ws, size(w)), atol=1e-20, rtol = 1e-15))
+        @test all(isapprox(c, partition_global_array(arch, cs, size(c)), atol=1e-20, rtol = 1e-15))
+        @test all(isapprox(η, partition_global_array(arch, ηs, size(η)), atol=1e-20, rtol = 1e-15))
```

Comment:
maybe call `partition_global_array(arch, us, size(u))` on a different line to make this easier to read